builder/virtualbox/common/step_attach_guest_additions.go

```diff
@@ -77,7 +77,7 @@ func (s *StepAttachGuestAdditions) Cleanup(state multistep.StateBag) {
 		"--medium", "none",
 	}
 
-	if err := driver.VBoxManage(command...); err != nil {
-		ui.Error(fmt.Sprintf("Error unregistering guest additions: %s", err))
-	}
+	// Remove the ISO. Note that this will probably fail since
```

Comment:
we're not setting `guest_additions_attached` to anything in particular, just using it as a sentinel value.

We do use the state bag to maintain state, but we don't mutate it, in general. Setting an additional flag for each device we've already tried to detach sounds fine, and will accept PRs for that.

I think the way that I'd like to see this done is to create a new class called something like `VBoxDevice` which holds all the data in the `storageattach` command. We'd set this in the state: `state.Put("guest_additions_device", &VBoxDevice{...})`.

We'd then add two methods to the driver: `AttachDevice` and `DetachDevice`. `DetachDevice` could consult a flag/`sync.once` object to make sure we only detached once.

I think this would be a fun thing to write, and would make the code simpler, but I don't have bandwidth. Will happily look at a PR if you make those changes.